### ISSUE TITLE
Dynamic Build Matrix

### ISSUE THREAD

**Issue by angerman** [input-output-hk/devx#101]
We currently wait for hydra jobs, and then run a set of pre-defined jobs to build. But it would be better (also ux wise) if we took the hydra jobs, and then generically generated the containers/codespaces. Maybe use a regex to extract the necessary bits, and create a dynamic matrix. 

https://www.kenmuse.com/blog/dynamic-build-matrices-in-github-actions/.

To get this right, it somewhat depends on #100.

**Comment by andreabedini** [comment]
We can use nix-eval-jobs for that.  IIRC can point it to a specific attribute to use as root.

**Comment by angerman** [comment]
@andreabedini yes, we just need to make sure we don't need to build anything 😅

**Comment by andreabedini** [comment]
> @andreabedini yes, we just need to make sure we don't need to build anything 😅

true :-/ Maybe `--option allow-import-from-derivation false` can be a fail safe?
